ee/clickhouse/models/property.py

```diff
@@ -153,9 +156,17 @@ def parse_prop_clauses(
                     f"{property_operator} 0 = 13"
                 )  # If cohort doesn't exist, nothing can match, unless an OR operator is used
             else:
-                person_id_query, cohort_filter_params = format_filter_query(cohort, idx)
-                params = {**params, **cohort_filter_params}
-                final.append(f"{property_operator} {table_name}distinct_id IN ({person_id_query})")
+
+                if person_properties_mode == PersonPropertiesMode.EXCLUDE:
```

Comment:
Yeah, a bit confusing. So, this test `test_entity_filtered_by_cohort` just as an event specific filter. Since the query builder will default to property group logic if `properties` doesn't exist, it was using the prop_group logic to build the event filter. However, there was some mismatch with how the query builder was parsing cohort properties ([here](https://github.com/PostHog/posthog/pull/8589/commits/713c76e79eae98ee7acdccf326dc48f42d6d17b6#diff-1459892830dd906ace86efd2bf27bed0fe3f3e2637772911be7c32e7f61c1a37L197)) vs how the parse_prop_clauses should handle cohort subqueries (the code this comment is looking at).  In other words, the eventquery class was shortcircuiting the logic for cohort subqueries with `_get_cohort_subquery` and handling things differently